src/Compilers/Core/Portable/Operations/IHasDynamicArgumentsExpression.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Semantics
+{
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IHasDynamicArgumentsExpression : IOperation
+    {
+        /// <summary>
+        /// Name of the dynamically invoked member.
+        /// </summary>
+        string Name { get; }
+
+        /// <summary>
+        /// List of applicable symbols that are dynamically bound to the <see cref="Name"/>.
+        /// </summary>
+        ImmutableArray<ISymbol> ApplicableSymbols { get; }
```

Comment:
This interface will also be implemented by [BoundDynamicIndexerAccess](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Generated/BoundNodes.xml.Generated.cs#5484), which has applicable property symbols.